### PR TITLE
Nullable fun

### PR DESCRIPTION
**Pull request by vijaye-statsig**
Using nullables to catch and fix potential nullref bugs

### CODE REVIEW

File: dotnet-statsig/src/Statsig/Server/Evaluation/ConfigCondition.cs [hunk 1]
```diff
@@ -29,19 +29,25 @@ internal ConfigCondition(string type, JToken targetValue, string op, string fiel
...
+                valuesMap = additionalValues.ToObject<Dictionary<string, object>>() ?? valuesMap;
+            }
             return new ConfigCondition(
-                type.Value<string>(),
+                type.Value<string>() ?? "",
                 jobj.TryGetValue("targetValue", out targetValue) ? targetValue : null,
                 jobj.TryGetValue("operator", out op) ? op.Value<string>() : null,
                 jobj.TryGetValue("field", out field) ? field.Value<string>() : null,
-                jobj.TryGetValue("additionalValues", out additionalValues) ? additionalValues.ToObject<Dictionary<string, object>>() : new Dictionary<string, object>(),
+                valuesMap,
```
Comment (jkw-statsig) [inline dotnet-statsig/src/Statsig/Server/Evaluation/ConfigCondition.cs:50]: why the change here?

**Review by jkw-statsig — APPROVED** [review]

File: dotnet-statsig/src/Statsig/Server/Evaluation/ConfigCondition.cs (hunk 1 above)
Comment (vijaye-statsig) [reply dotnet-statsig/src/Statsig/Server/Evaluation/ConfigCondition.cs:50]: Because ToObject could be null as well and it is awkward to write all that in a single statement.
